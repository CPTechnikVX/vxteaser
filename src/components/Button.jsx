import React from 'react';
import styled from 'styled-components';

class Banner extends React.PureComponent {
	render() {
		const Button = styled.a`    
			border-radius: 2px;
		    display: inline-block;
		    font-family: 'Roboto';
		    cursor: pointer;
		    min-width: 50px;
		    position: relative;
		    text-align: left;
		    text-decoration: none;
		    transition: all 0.15s linear;
		    vertical-align: top;
		    padding: 0.7em 5em 0.65em;
		    font-size: 1.7rem;
		    background-color: #1F93E9;
		    border: 1px solid #1F93E9;
		    color: #FFFFFF;
		    text-transform: none;
		`;

		return <Button onClick={this.props.onClick}>{this.props.children}</Button>;
	}
}

export default Banner;