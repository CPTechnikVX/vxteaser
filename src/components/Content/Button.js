import React      from 'react';
import PropTypes  from 'prop-types';
import Constants  from '../../utils/Constants';
import classnames from 'classnames';

/**
 * General button element
 */
export default class Button extends React.PureComponent {
	render() {
		const {children, color, link} = this.props;
		const classList               = [];
		let onClickFn;

		classList.push(Constants.ClassName.Button);

		if (color) {
			classList.push(Constants.PrefixClassName.Color + color);
		}

		if (link) {
			onClickFn = (e) => {
				e.preventDefault();
				e.stopPropagation();

				this.props.onLinkFn(link);
			};
		}

		return <a className={classnames(classList)} onClick={onClickFn}>{children}</a>;
	}
}

Button.propTypes = {
	/** @ignore */
	children: PropTypes.node,
	/**
	 * Textual color definition
	 */
	color:    PropTypes.oneOf(['primary']),
	link:     PropTypes.string,
	/** @ignore */
	onLinkFn: PropTypes.func,
};
