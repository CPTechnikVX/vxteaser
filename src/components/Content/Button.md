```js
<Button color="primary" link="https://www.example.com">Jetzt spielen!</Button>
```