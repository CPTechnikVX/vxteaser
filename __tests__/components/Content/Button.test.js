import React    from 'react';
import {render} from '@testing-library/react';
import Button   from '../../../src/components/Content/Button';

describe('render button', () => {
    test('empty', () => {
        const {container} = render(
                <Button />
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    test('with props', () => {
        const {container} = render(
                <Button color="primary" link="https://www.example.com">This is button</Button>
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
