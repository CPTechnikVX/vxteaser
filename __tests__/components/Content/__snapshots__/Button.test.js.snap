// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button empty 1`] = `
<a
  class="vxteaser-button"
/>
`;

exports[`render button with props 1`] = `
<a
  class="vxteaser-button vxteaser-bg-color--primary"
>
  This is button
</a>
`;
